packages/elements/src/components/file-picker/file-picker.component.ts

```diff
@@ -232,49 +256,44 @@ export class MutationTestReportFilePickerComponent extends LitElement {
     this.#closePicker();
   }
 
-  #togglePicker(event: KeyboardEvent | null = null) {
+  #togglePicker = (event: KeyboardEvent | null = null) => {
     event?.preventDefault();
     event?.stopPropagation();
 
     this.openPicker = !this.openPicker;
+  };
 
-    if (!this.openPicker) {
-      document.body.style.overflow = 'auto';
-    } else {
-      document.body.style.overflow = 'hidden';
-    }
-
-    void this.updateComplete.then(() => this.#focusInput());
-  }
-
-  #focusInput() {
-    this.shadowRoot?.querySelector('input')?.focus();
-  }
+  #focusInput = () => {
+    this.renderRoot.querySelector('input')?.focus();
+  };
 
-  #closePicker() {
-    document.body.style.overflow = 'auto';
+  #closePicker = () => {
     this.openPicker = false;
     this.fileIndex = 0;
     this.#filter('');
-  }
+  };
 
-  #handleSearch(event: KeyboardEvent) {
+  #handleSearch = (event: InputEvent) => {
     if (!this.openPicker) {
       return;
     }
 
-    if (event.key === 'ArrowDown' || event.key === 'ArrowUp' || event.key === 'Tab') {
-      return;
-    }
-
     this.#filter((event.target as HTMLInputElement).value);
     this.fileIndex = 0;
-  }
+  };
 
   #filter(filterKey: string) {
-    this.filteredFiles = Array.from(this.#searchMap.keys())
-      .filter((file) => file.toLowerCase().includes(filterKey.toLowerCase()))
-      .map((file) => ({ name: file, file: this.#searchMap.get(file)! }));
+    if (!filterKey) {
+      this.filteredFiles = this.#searchTargets;
+    } else {
+      this.filteredFiles = fuzzysort.go(filterKey, this.#searchTargets, { key: 'prepared', threshold: 0.3, limit: 500 }).map((result) => ({
```

Comment:
Question: the limit here indicates the time? or is it something else?

Reply:
It's the number of search results. Not super common to have > 500 files, but it is recommended to keep the search fast. Search was always around 1ms for me in the large example